evennia/objects/objects.py

```diff
@@ -1597,6 +1609,43 @@ def at_cmdset_get(self, **kwargs):
         """
         pass
 
+    def get_object_typeclass(self, obj_type: str = "object", typeclass: str = None, method: str = "create", **kwargs) -> tuple[
```

Comment:
I don't think this makes much sense to have on the typeclass like this. 
- You are accessing a method _on_ a typeclass instance in order to ... get another typeclass instance. While intended for build commands, this method's name reflects nothing of the sort; it's named as a generic method for getting typeclasses. 
- Moreover, you are intending to use this on the caller (which is not clear from this docstring), essentially making this worthless, or at least with no clear usage on other typeclassed entities. 

This really doesn't belong here. It should go in the build command parent; you lose no flexibility this way compared to the current setup since you'd only customize this on the `caller` typeclass anyway.

Reply:
Putting it in the build command parent makes it far harder to overload without then needing to re-create subclasses of all of the build command children... I am very much attempting to extricate this logic out of the commands and onto something independent of the commands.

Putting it on the object that's doing the building made sense, because that would have access to its location and other relevant information about who's doing the building...